joomla/joomla-cms
# Tab state better handling of multiple states

Pull Request for Issue #15710 .

### Summary of Changes

better handling of multiple tab widgets per page (by full xpath) and not saving the state if there is no id in the url (like on the CREATE page of content. giving better future proofing as well



### Testing Instructions

Edit Article 1 - click a tab - click save - same tab should be selected
Edit Article 1 - click PERMISSIONS tab, Click an ACL Level - click save - same TABS should be selected
SAVE and CLOSE Article 1 
Edit Article 2 - First Tab should be selected 
SAVE and CLOSE Article 2
Create new article - NOTE first tab is selected.
Select the LAST tab, Click cancel
Create new article - NOTE first tab is selected.
Edit Article 1 - click the LAST TAB
Click cancel - go to installer and see FIRST TAB correctly shown and loaded **(bonus fix!!!)** 

### Two requested NEW features that this PR contains

Ability to preserve state from CREATE to EDIT on first save.
Ability to link directly to a tabs #anchor using the hash of a url and have that tab be active on page load
